examples/dcutr/README.md

```diff
@@ -0,0 +1,42 @@
+## Description
+
+The "Direct Connection Upgrade through Relay" (DCUTR) protocol allows peers in a peer-to-peer network to establish direct connections with each other.
+This example provides a basic usage of this protocol in **libp2p**.
+
+## Usage
+
+To run the example, follow these steps:
+
+1. Run the example using Cargo:
+   ```sh
+   cargo run -- <OPTIONS>
+   ```
+   Replace `<OPTIONS>` with specific options.
+
+### Available Options
+
+The following options can be used with the example:
+
+- `--mode <MODE>`: Specify the mode of operation for the example. Possible values are `dial` (client-dial) or `listen` (client-listen).
+- `--secret-key-seed <SEED>`: Specify a fixed value to generate a deterministic peer ID.
+- `--relay-address <ADDRESS>`: Specify the listening address for the relay server.
+- `--remote-peer-id <ID>`: Specify the Peer ID of the remote peer to establish a connection with (only in `dial` mode).
+
+### Example usage
+
+- Example usage in client-dial mode:
+	```sh
+	cargo run -- --mode dial --secret-key-seed 42 --relay-address /ip4/127.0.0.1/tcp/12345 --remote-peer-id <REMOTE_PEER_ID>
+	```
+
+- Example usage in client-listen mode:
+	```sh
+	cargo run -- --mode listen --secret-key-seed 42 --relay-address /ip4/127.0.0.1/tcp/12345
+	```
```

Comment:
For this to work, we need a relay server too. Can we add the instructions for that? Plus, the listener should be started first.

Reply:
@thomaseizinger I just added the ref giving the link to the `examples/relay-server` folder to avoid giving here an instruction that is already somewhere else.